### PR TITLE
Set error on LINGER and DONTLINGER socket options

### PR DESCRIPTION
**Pull request by mortezag**
As linger and dontlinger options were hacked into previous lwip stack and those hacks do not exist in the new lwip options, calling setsockopt and getsockopt with these values will cause problems. This fix returns proper errors for them.

### CODE REVIEW

File: DeviceCode/pal/lwip_1_4_1_os/SocketsDriver/lwIP__Sockets.cpp [hunk 1]
```diff
@@ -562,13 +562,18 @@ int LWIP_SOCKETS_Driver::SetSockOpt( SOCK_SOCKET socket, int level, int optname,
             nativeOptionName = GetNativeSockOption(optname);            
 
             switch(optname)
-            {
+            {        
+                // LINGER and DONTLINGER are not implemented in LWIP
+                case SOCK_SOCKO_LINGER:
+                case SOCK_SOCKO_DONTLINGER:
+                    errno = SOCK_ENOPROTOOPT;
+                    return SOCK_SOCKET_ERROR;
```
Comment (smaillet-ms) [inline DeviceCode/pal/lwip_1_4_1_os/SocketsDriver/lwIP__Sockets.cpp]: Any chance we could use a more descriptive error code! This smells a bit like E_FAIL... (same for other cases in here)

Comment (mortezag) [inline DeviceCode/pal/lwip_1_4_1_os/SocketsDriver/lwIP__Sockets.cpp]: Maybe I didn't get your point but the error code is pretty specific. The error returned to the user is SOCK_ENOPROTOOPT which means "Bad protocol option: An unknown, invalid or unsupported option or level was specified in a getsockopt or setsockopt call." Unfortunately, it is not possible to return a error message to the user.

**Comment by smaillet-ms** [comment]
OK, I guess I misread the code. the errno is what is exposed to the user, I was concerned about the SOCK_SOCKET_ERROR return value (Which is basically a generic E_FAIL) since that's not really the error code that percolates up to the caller it's effectively a success/fail so no worries.

**Comment by smaillet-ms** [comment]
sorry should have include :+1:

**Pull request merged**